### PR TITLE
Publish criteria to the web

### LINKED ISSUE #38: Publish criteria to web
As we discussed in the [last meeting](https://docs.google.com/document/d/16tL1Ln7owIRXSoCKgyYHCs9-JP9iw-ouyk8koGAeHA0/edit?tab=t.0#heading=h.inzqutwp94f8), we should publish the criteria to a website instead of rendering markdown from the yaml and committing the result.

I volunteered to work on a proof of concept for this, so I'm opening an issue to track the work.

- [x] Build the workflow (#81)
- [x] Enable Pages (#83)
- [x] Get CNAME (#82)
- [x] Configure the CNAME in the repo settings (and then fix whatever we discover doesn't work) (#85)

### PR DESCRIPTION
**Pull request by funnelfiasco**
Automatically publish the criteria to GitHub pages instead of rendering yaml to markdown and committing the result to the repo.

Highlights include

* A GitHub workflow to publish the content
* A modified theme to update the default "improve this page" link
* Added a table of contents at the top of the page for easier navigation
* Minor wording and formatting changes to `template.md` to make it work better as a web page
* Addition of a link to the contributors in the acknowledgements section

Implements #38

(*Edited to say that it implements 38, not fixes, because we'll want to keep that issue open to track getting a DNS record as well*)

### CODE REVIEW

**Comment by funnelfiasco** [comment]
Marked as a work in progress for the moment because I've left my branch in the workflow so that I can easily publish updates based on comments. I'll fix that before the final merge.

Preview at https://funnelfiasco.github.io/security-baseline/

**Comment by eddie-knight** [comment]
No change requests from me at this time. This looks like a really useful and practical solution.

File: .github/workflows/web-publish.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+            persist-credentials: false
+      - name: Setup Pages
+        uses: actions/configure-pages@983d7736d9b0ae728b81ab479565c72886d7745b # v5.0.0
+      - name: Build content from yaml
+        run:  cd cmd && go run main.go compile --output ../docs/index.md
```
Comment (puerco) [inline .github/workflows/web-publish.yml:34]: Do you think this step needs [actions/setup-go](https://github.com/actions/setup-go) first?

**Comment by SecurityCRob** [comment]
me likey, tyvm

**Comment by funnelfiasco** [comment]
Updated the branch to use `main` and marked it as ready for review.

**Review by eddie-knight — APPROVED** [review]

File: .github/workflows/web-publish.yml (hunk 1 above)
Comment (puerco) [reply .github/workflows/web-publish.yml:34]: sounds good! TIL go is in the runner :)

**Pull request merged**